### DISCUSSION TITLE
"Permission denied" when uninstalling Netdata

### DISCUSSION THREAD

**Discussion by arno56** [netdata/netdata#13923] [Q&A]
Hi,

I am trying to uninstall Netdata from my server via your official netdata-kickstart.sh and I am getting the message error bellow.

I tried to play with permissions and such but it doesn't change anything.

`[root@ds1*** [~]]# wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh --uninstall
--2022-11-01 08:12:44--  https://my-netdata.io/kickstart.sh
Resolving my-netdata.io (my-netdata.io)... 188.114.96.0, 188.114.97.0, 2a06:98c1:3120::, ...
Connecting to my-netdata.io (my-netdata.io)|188.114.96.0|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/octet-stream]
Saving to: ‘/tmp/netdata-kickstart.sh’

    [ <=>                                                                                 ] 78,871      --.-K/s   in 0.002s

2022-11-01 08:12:44 (34.4 MB/s) - ‘/tmp/netdata-kickstart.sh’ saved [78871]

/tmp/netdata-kickstart.sh: line 519: /tmp/netdata-test.BbaV22FHI9: Permission denied
 --- Using /tmp/netdata-kickstart-rGlP2j5T4J as a temporary directory. ---
 --- Checking for existing installations of Netdata... ---
 --- Downloading netdata-uninstaller ... ---
[/tmp/netdata-kickstart-rGlP2j5T4J]# curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-rGlP2j5T4J/netdata-uninstaller.sh https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/netdata-uninstaller.sh
 OK

[/tmp/netdata-kickstart-rGlP2j5T4J]# /tmp/netdata-kickstart-rGlP2j5T4J/netdata-uninstaller.sh --yes
/tmp/netdata-kickstart.sh: line 465: /tmp/netdata-kickstart-rGlP2j5T4J/netdata-uninstaller.sh: Permission denied
 FAILED

Finished uninstalling the Netdata Agent.The following non-fatal warnings or errors were encountered:

  - Command "/tmp/netdata-kickstart-rGlP2j5T4J/netdata-uninstaller.sh --yes" failed with exit code 126.

[root@ds1*** [~]]#`

**Comment by MrZammler** [comment]
Hi @arno56 ! What system is this?

Can you try to use another tmp directory by using $TMPDIR? I.e. `TMPDIR=/another/directory/ && wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh --uninstall` ?

**Reply by arno56** [reply]
Hi [MrZammler](https://github.com/MrZammler), x86_64 GNU/Linux

I've just tried your solution but I still get the same result.

**Reply by MrZammler** [reply]
Hmm, can you also try to download `netdata-kickstart.sh` to the same dir as TMPDIR?

i.e. `TMPDIR=/another/directory/ && wget -O /another/directory/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /another/directory/netdata-kickstart.sh --uninstall`

**Reply by arno56** [reply]
Ahhhhh, still no success.

`[root@ds1*** [~]]# TMPDIR=/tempnetdata/ && wget -O /tempnetdata/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tempnetdata/netdata-kickstart.sh --uninstall
--2022-11-01 09:27:15--  https://my-netdata.io/kickstart.sh
Resolving my-netdata.io (my-netdata.io)... 188.114.96.0, 188.114.97.0, 2a06:98c1:3120::, ...
Connecting to my-netdata.io (my-netdata.io)|188.114.96.0|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/octet-stream]
Saving to: ‘/tempnetdata/netdata-kickstart.sh’

    [ <=>                                                                                 ] 78,871      --.-K/s   in 0.009s

2022-11-01 09:27:15 (8.48 MB/s) - ‘/tempnetdata/netdata-kickstart.sh’ saved [78871]


/tempnetdata/netdata-kickstart.sh: line 519: /tmp/netdata-test.ENb04DbMDZ: Permission denied
 --- Using /tmp/netdata-kickstart-ldqooBdVak as a temporary directory. ---
 --- Checking for existing installations of Netdata... ---
 --- Downloading netdata-uninstaller ... ---
[/tmp/netdata-kickstart-ldqooBdVak]# curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-ldqooBdVak/netdata-uninstaller.sh https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/netdata-uninstaller.sh
 OK

[/tmp/netdata-kickstart-ldqooBdVak]# /tmp/netdata-kickstart-ldqooBdVak/netdata-uninstaller.sh --yes
/tempnetdata/netdata-kickstart.sh: line 465: /tmp/netdata-kickstart-ldqooBdVak/netdata-uninstaller.sh: Permission denied
 FAILED

Finished uninstalling the Netdata Agent.The following non-fatal warnings or errors were encountered:

  - Command "/tmp/netdata-kickstart-ldqooBdVak/netdata-uninstaller.sh --yes" failed with exit code 126.

[root@ds1*** [~]]#`

**Reply by MrZammler** [reply]
Can you please try with adding export ? `export TMPDIR=/tempnetdata/ && wget -O /tempnetdata/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tempnetdata/netdata-kickstart.sh --uninstall` ?

**Comment by arno56** [comment]
Even with the TMPDIR= command, it keeps using the default '/tmp/  directory 
2022-11-01 08:34:00 (25.8 MB/s) - ‘/tmp/netdata-kickstart.sh’ saved [78871]

**Comment by awsto** [comment]
Try to run  /bin/bash /tempnetdata/netdata-kickstart.sh --y
You will need the environment file at: /etc/netdata/.environment
About env: https://learn.netdata.cloud/docs/agent/packaging/installer/uninstall
